CHT/Temperature.C

```diff
@@ -56,6 +56,21 @@ void preciceAdapter::CHT::Temperature::write(double* buffer, bool meshConnectivi
     }
 }
 
+void preciceAdapter::CHT::Temperature::writeGradients(std::vector<double>& gradientBuffer, const unsigned int dim)
+{
+    for (const label patchID : patchIDs_)
+    {
+        const auto& TGrad = fvc::grad(*T_);
+        forAll(TGrad().boundaryField()[patchID], i)
+        {
+            for (unsigned int d = 0; d < dim; ++d)
+            {
+                gradientBuffer[i * dim + d] = TGrad().boundaryField()[patchID][i][d];
+            }
+        }
+    }
+}
```

Comment:
This essentially addresses the same problem as @thesamriel implemented in `FF/TemperatureGradient.C` in https://github.com/precice/openfoam-adapter/pull/281/files#diff-90c837067b5e748c5bc1af9d24a5ab6f0a5f7801bd91685900c27491c259a830

The main difference is that:

- This PR gives the gradient of (temperature) to preCICE, as gradient data
- #281 gives the (gradient of temperature) to preCICE, as normal data values

Would these two PRs need to be aligned somehow? @thesamriel I assume that this still cannot be shaped as a replacement for the respective addition of #281.